1st code

dzu181 opened the pull request:
The first version - only chest() work.
And then game is over.
"is_game_continue" variable is useless in this code.

dzu181 committed:
1st code

The first version - only chest() work.
And then game is over.
"is_game_continue" variable is useless in this code.

dzu181 committed:
Add files via upload

This version doesn't use is_game_continue in start().
Also no while loop in start().

dzu181 committed:
Update ex36.py

Try to update file to see history change of files on Github.

dzu181 committed:
Update ex36 - without is_game_continue.py

Update: 
- Put every choice menu under 'while' loop to remind the player about available choices.
- Added some additional comments.

dzu181 committed:
Update ex36 - without is_game_continue.py

Rewrite some comments part to create TODO list, English comments all code.

dzu181 committed:
Update ex36 - without is_game_continue.py

Hide sensitive detail which spoils the storyline.

dzu181 committed:
Update ex36 - without is_game_continue.py

Changed repeated print() into individual functions.

dzu181 committed:
Update ex36 - without is_game_continue.py

Modified great_wall() a bit.

dzu181 committed:
Update ex36 - without is_game_continue.py

Added 3rd choice to the great_wall() - "3. Shout to the sky".

dzu181 committed:
Update ex36 - without is_game_continue.py

- Changed Crow_false_attempt limit to 2.
- Scare the crow no longer increaes Crow_false_attempt

altsun commented on ex36.py:
Remove this file if it is no longer maintained

dzu181 replied:
File deleted. Thank you! :)

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,206 @@
+
+
+def start():
+	print("You don't remember anything. You are fallen.")
```
Create a function, e.g. show_introduction(), for 3 prints
Because the main function start() should be the most abstract, so it should not be too long and include too many low level statements

dzu181 replied:
Ok. How about I apply all the introduction lines of all functions (chest() , crow(), great_wall(), etc. ), into smaller print functions? Will it be clean code?

altsun replied:
No need for now. The function start() should be the most abstract. Other functions can have low level statements

altsun commented on ex36 - without is_game_continue.py:
Please don't use tabs for indentation, use spaces instead

dzu181 replied:
For this beginning level project, I will allow myself to use tabs.

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,206 @@
+	chest_key_obtained = False
+
+	while not chest_key_obtained:
+		## Remind player of choices
```
I think this comment is redundant, the name of the function show its meaning

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,206 @@
+	print("Alright! Let's open this chest.")
+	
+	while True:
+		## Remind player of choices
```
I think this comment is redundant, the name of the function show its meaning

dzu181 replied:
Got it.

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,206 @@
+		open_choice = input("> ")
+	
+		if open_choice == "1":
+			print("\n================================================")
```
This long line is repeated everywhere, use a constant for it or put the print statement in a function

dzu181 replied:
I think this can be deleted from code.
We don't have a proper UI for this project anyway.

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,206 @@
+			return True
+
+# The Great Wall
+def great_wall(chest_key, crow_key):
```
Add a sentence to tell player that he continue moving from the crow tree

dzu181 replied:
I think the sentence will be more suitable putting in the later function : great_wall(), to show up at the beginning of the function.

altsun replied:
test

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,206 @@
+
+		# Player rage mode. Destroy things!	(wall_attempt == 3)		
+		wall_attempt +=1
+		if wall_attempt == 3:
```
3 is a magic number
Create a const like:
```python
MAX_WALL_ATTEMPT = 3
if wall_attempt == MAX_WALL_ATTEMPT:
```
After this you can remove `(wall_attempt == 3)` in the above comment

dzu181 replied:
Got it.

dzu181 committed:
Delete ex36.py

Code file no longer used.
Will be archived for the future.

dzu181 committed:
Update ex36 - without is_game_continue.py

- Replaced magic number 3 with a variable.
- Added a sentence to tell player that he continue moving from the crow tree to the Great wall. (in great_wall() ).
- Deleted long repetitive lines (==========).
- Deleted unnecessary comments (repetitive, redundant).
- Created a new first_intro() function for a short start().

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -126,9 +109,9 @@ def great_wall(chest_key, crow_key):
 	
 	# A surprise later - hidden choice
 	wall_attempt = 0
+	max_wall_attempt = 3
```
Use ALL_CAPS convention for const

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -166,18 +149,21 @@ def kick_door():
 
 # A more painful dead.
 def count_mistake(number):	
 	if number >= 2:
```
Define a const here for MAX_MISTAKE

dzu181 committed:
Update ex36 - without is_game_continue.py

- Updated kick_door() function.
- Added some crazy choices in kick_door().

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,217 @@
+			print("But no one can help you now. Better be back.")
+
+
+		# Player rage mode. Destroy things!	(wall_attempt == 3)		
```
Move this comment above the if statement

altsun replied:
Above the line 143

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,217 @@
+		great_wall_choice = input("> ")
+
+		if great_wall_choice == "1":
+			if chest_key:
```
chest_key is always true, right?

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,217 @@
+		print("2. ☌⟟⎐⟒ ⎍⌿")
+		print("3. Give up.")
+
+		kicking = input("> ")
```
Why don't rename this variable to a simple and meaningful, like `choice`

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,217 @@
+	exit(0)
+
+# Begin game
+start()
```
Move this main function to top

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,217 @@
+	print("You decide to stomp your feet towards the door until it cracks.")
+
+	for wall_hp in range(100,0,-1):
+		print("The wall's HP is:", wall_hp, "\n")
```
I think this is a hidden action. So it is better to hide wall hp?

altsun committed:
add function display_text

altsun committed:
add const MENU_DELAY

altsun committed:
fix display of dead reason

altsun committed:
improve display_text with multiple args

altsun committed:
optimize

altsun committed:
create display_utils

altsun committed:
remove redundant spaces

altsun committed:
remove unused import

dzu181 committed:
Merge pull request #2 from dzu181/feature/display_text

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,228 @@
+			
+		elif chest_choice == "2":
+			chest_key_obtained = open_chest()
+		else:
```
Fix this to chest_choice == "3"
Add another else to repeat the menu or tell player to input the correct choice

dzu181 committed:
Update ex36 - without is_game_continue.py

- Fixed a minor bug in line 38 and 171, that forces player to out game unintentionally.

dzu181 committed:
Update ex36 - without is_game_continue.py

- Undo 1 hot fix at line 171, and change line 158 content to better fit the for loop.

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,228 @@
+	display_text("You got super angry!! You tried anything but the door won't open!")
+	display_text("You decide to stomp your feet towards the door until it cracks.")
+
+	for wall_hp in range(100,0,-1):
```
Change to while loop

altsun replied:
Change to while loop

altsun commented on ex36 - without is_game_continue.py:
```diff
@@ -0,0 +1,228 @@
+		display_text("Wanna kick the damn door?")
+		display_text("1. Hell yeah!")
+		display_text("2. ☌⟟⎐⟒ ⎍⌿")
+		display_text("Any other key: Give up.")
```
Remember to fix this

dzu181 committed:
Update ex36 - without is_game_continue.py

- Replace for loop in kick_door() with while loop.
- Make a delay in "check pocket" action in great_wall.
- Fix the else in kick_door loop.

dzu181 committed:
Update display_utils.py

- Shorten the delay time for easier re-play game.

altsun committed:
Merge branch 'main' into dzu181-patch-1
